SpotifyAPI/SpotifyAPI.ino

```diff
@@ -0,0 +1,233 @@
+           String imageExtension = albumArtURL.substring(albumArtURL.indexOf("/", beginningIndex));
+
+           if(SSLClient.connectSSL(imageApi.c_str(), CONNECTION_PORT)) {
+               SSLClient.println(GET + imageExtension + " " + HTTP_VERSION);
+               SSLClient.println(HOST + imageApi);
+               SSLClient.println(AUTHORIZATION + "Bearer " + oauthToken);
+               SSLClient.println(CONNECTION_CLOSE);
+               SSLClient.println();
+           }
+       }
+
+       // Checking methods
+       bool checkHTTPStatus() {
+           char status[32] = {0};
+           SSLClient.readBytesUntil('\r', status, sizeof(status));
+           if (strcmp(status, "HTTP/1.1 200 OK") != 0) {
+               Serial.print("Unexpected response: ");
+               Serial.println(status);
+               SSLClient.stop();
+               return false;
+           }
+           return true;
+       }
+
+       bool skipHTTPHeaders() {
+           char endOfHeaders[] = "\r\n\r\n";
+           if (!SSLClient.find(endOfHeaders)) {
+               Serial.println("Invalid response");
+               SSLClient.stop();
+               return false;
+           }
+       }
+
+    
+    WiFiSSLClient SSLClient;
+
+    String oauthToken;
+    long oauthExpiryTime = 0;
+
+    long nextAlbumArtRetrievalTime = 0;
+    long nextDisplayUpdate = 0;
+    String albumArtURL;
+
+    // Wifi Constants
+    const uint16_t CONNECTION_PORT = 443;
+    String GET = "GET";
+    String POST = "POST";
+    String HOST = "Host: ";
+    String AUTHORIZATION = "Authorization: ";
+    String HTTP_VERSION = "HTTP/1.1";
+    String CONTENT_TYPE = "Content-Type: ";
+    String CONTENT_LENGTH = "Content-Length: ";
+    String CONNECTION_CLOSE = "Connection: close";
+};
+
+class WiFiUtilities {
+
+    public:
+        WiFiUtilities(String ssid, String password) : ssid(ssid), password(password) {};
+
+        void checkWiFiModule() {
+            if (WiFi.status() == WL_NO_MODULE) {
+                Serial.println("Communication With WiFi Module Failed!");
+                while (true)
+                ;
+            }
+
+            String firmwareVersion = WiFi.firmwareVersion();
+            if (firmwareVersion < WIFI_FIRMWARE_LATEST_VERSION) {
+                Serial.println("Please Upgrade the Firmware");
+            }
+        }
+
+        void connectToWiFi() {
+            Serial.println("Connecting to Wifi");
+
+            WiFi.begin(ssid.c_str(), password.c_str());
+            while (WiFi.status() != WL_CONNECTED) {
+                delay(200);
+            }
+
+            Serial.print("Connected to Wifi Network: ");
+            Serial.println(ssid.c_str());
+        }
+
+        String ssid, password;
+};
+
+
+// SETUP & LOOP
+void setup() {
+  Serial.begin(9600);
+  while (!Serial) {
+    ;  // Wait For Serial Port to Connect.
+  }
+  WiFiUtilities wifiCheck("The Force", "thisistheway");
```

Comment:
Maybe we should declare these first as const strings or something. Also maybe we should genericify this code and then import a secrets file or something.

Reply:
I thought so too. Same with the Spotify credentials? or maybe like a terminal getline for wifi and spotify creds??

Reply:
I think we should do the secrets file thing. It will act like a `.env` file.